components/collector/src/source_collectors/gitlab/base.py

```diff
@@ -143,29 +160,28 @@ async def _api_url(self) -> URL:
         return await self._gitlab_api_url(f"pipelines?updated_after={lookback_date}")
 
     async def _landing_url(self, responses: SourceResponses) -> URL:
-        """Override to return a landing URL for the most recent pipeline."""
-        urls = []
+        """Override to return a landing URL for the first pipeline, if there's one."""
+        if pipelines := await self._pipelines(responses):
+            return URL(pipelines[0].web_url)
+        return await super()._landing_url(responses)
+
+    async def _pipelines(self, responses: SourceResponses) -> list[Pipeline]:
+        """Get the pipelines from the responses."""
+        pipelines = []
         try:
             for response in responses:
-                pipelines = await self._pipelines(response)
-                urls.extend([(self._datetime(pipeline), pipeline["web_url"]) for pipeline in pipelines])
+                json = await response.json()
+                # The GitLab pipelines endpoint returns a list or, if a pipeline id is passed, one pipeline. Harmonize:
+                json_list = json if isinstance(json, list) else [json]
+                pipelines.extend([Pipeline(**pipeline) for pipeline in json_list])
         except StopAsyncIteration:
             pass
-        return URL(max(urls, default=(None, await super()._landing_url(responses)))[1])
-
-    async def _pipelines(self, response: Response) -> list[Pipeline]:
-        """Get the pipelines from the response."""
-        return [pipeline for pipeline in await response.json() if self._include_pipeline(pipeline)]
+        return [pipeline for pipeline in pipelines if self._include_pipeline(pipeline)]
 
     def _include_pipeline(self, pipeline: Pipeline) -> bool:
         """Return whether this pipeline should be considered."""
-        return (
-            pipeline["ref"] == self._parameter("branch")
-            and pipeline["status"] in self._parameter("pipeline_statuses_to_include")
-            and pipeline["source"] in self._parameter("pipeline_triggers_to_include")
-        )
-
-    @staticmethod
-    def _datetime(pipeline: Pipeline) -> datetime:
-        """Return the datetime of the pipeline."""
-        return parse_datetime(pipeline.get("updated_at") or pipeline["created_at"])
+        branches = self._parameter("branches")
+        matches_branches = match_string_or_regular_expression(pipeline.ref, branches) if branches else True
```

Comment:
Was there no exclude branch option?

Reply:
Nope. Let's wait to see if users need it.